src/Nox.Types/Types/Temperature/Temperature.cs

```diff
@@ -90,34 +63,38 @@ internal override ValidationResult Validate()
     }
 
     /// <summary>
-    /// Converts the this instance to its equivalent string representation using InvariantCulture format.
+    /// Converts <see cref="Temperature.TemperatureValue"/> to value in <see cref="TemperatureUnit.Celsius"/>.
     /// </summary>
-    /// <returns>A string representation of the <see cref="Temperature"/> object.</returns>
-    public override string ToString()
-        => $"{TemperatureValue.ToString($"0.{new string('#', 2)}", CultureInfo.InvariantCulture)} {Unit.ToSymbol()}";
-
-    /// <summary>
-    /// Converts <see cref="Temperature.TemperatureValue"/> to value in <see cref="TemperatureTypeUnit.Celsius"/>.
-    /// </summary>
-    /// <returns>TemperatureValue in <see cref="TemperatureTypeUnit.Celsius"/></returns>
-    public float TemperatureValueToCelsius()
+    /// <returns>TemperatureValue in <see cref="TemperatureUnit.Celsius"/></returns>
+    public QuantityValue ToCelsius()
     {
-        if (Unit == TemperatureTypeUnit.Celsius) return TemperatureValue;
+        if (Unit == TemperatureUnit.Celsius)
+            return Value;
+
+        var newObject = FromCelsius(((Value - 32) * 5) / 9);
 
-        return Round((TemperatureValue - 32) * 5 / 9);
+        return newObject.Value;
     }
 
     /// <summary>
-    /// Converts <see cref="Temperature.TemperatureValue"/> to value in <see cref="TemperatureTypeUnit.Fahrenheit"/>.
+    /// Converts <see cref="Temperature.TemperatureValue"/> to value in <see cref="TemperatureUnit.Fahrenheit"/>.
```

Comment:
You removed the property TemperatureValue and didn't update the comment.

Reply:
done